code/weapon/weapon.h

```diff
@@ -448,6 +448,11 @@ typedef struct weapon_info {
 	int				targeting_priorities[32];
 	int				num_targeting_priorities;
 
+	// Optional weapon failures
+	float failure_rate;
+	char failure_sub_name[NAME_LENGTH];
```

Comment:
This should be an `SCP_string` unless there is a very good reason to use a raw `char` array.